lxd/auth/bearer/bearer.go

```diff
@@ -22,43 +23,57 @@ func IsDevLXDRequest(r *http.Request, clusterUUID string) (isRequest bool, token
 		return false, "", ""
 	}
 
+	subject, _, err := isLXDToken(token, clusterUUID, encryption.DevLXDAudience(clusterUUID))
+	if err != nil {
+		return false, "", ""
+	}
+
+	return true, token, subject
+}
+
+// isLXDToken checks if the given token looks like it was issued by this LXD cluster and returns an error if it doesn't.
+// It does not verify the token signature.
+func isLXDToken(token string, clusterUUID string, expectedAudience string) (string, *time.Time, error) {
 	// Check we can parse the token. If we can't parse it, it could be an opaque OAuth2.0 token.
 	claims := jwt.MapClaims{}
 	t, _, err := jwt.NewParser().ParseUnverified(token, claims)
 	if err != nil {
-		return false, "", ""
+		return "", nil, errors.New("Token is not a JWT")
 	}
 
 	// There must be an issuer
 	issuer, err := t.Claims.GetIssuer()
 	if err != nil {
-		return false, "", ""
+		return "", nil, errors.New("Token does not have an issuer")
 	}
 
 	// There must be a subject
 	sub, err := t.Claims.GetSubject()
 	if err != nil {
-		return false, "", ""
+		return "", nil, errors.New("Token does not have a subject")
 	}
 
 	// Expect the issuer to be "lxd:{cluster_uuid}".
 	expectIssuer := encryption.Issuer(clusterUUID)
 	if issuer != expectIssuer {
-		return false, "", ""
+		return "", nil, errors.New("Token issuer does not match")
 	}
 
-	// Expect the audience to be "devlxd:{cluster_uuid}".
-	expectAudience := encryption.DevLXDAudience(clusterUUID)
 	audience, err := t.Claims.GetAudience()
 	if err != nil {
-		return false, "", ""
+		return "", nil, errors.New("Token does not have an audience")
```

Comment:
Same point as @kadinsayani made above, why are we losing the original error here?

Reply:
This is now wrapped